### ISSUE TITLE
Suggestion related to your now closed Code Review question on the even() method

### ISSUE THREAD

**Issue by holroy** [rsp/ppj#1]
It's been a while since I dabbled in Java, but if you don't care that much on memory size you could use the first half of this code, and if you need for the returned array to be of exact size use all of it:

```
static int[] even(int[] source) {
    int even_count;

    // Loop and calculate remainder only once, into tmp array
    int[] tmp = new int[source.length];
    for (i = even_count = 0; i < source.length; i++) {
        if (source[i] % 2 == 0) {
           tmp[even_count++] = source[i];
        }
    }

    // Make exact size copy
    int[] result = new int[even_count];
    System.arraycopy(tmp, 0, result, 0, evencount);

    return r;
}
```

Code is untested, but should work. I think. And use braces, not one-liners which hinders readability a lot. Better naming is also good.

**Comment by holroy** [comment]
This is not really an issue, but as the question got closed, and I had nearly finished my answer, I wanted it show somewhere...

**Comment by rsp** [comment]
@holroy Thanks a lot. I was really surprised that this question got negative votes and was closed as off-topic. Having written a lot of answers on Stack Overflow, I know how frustrating it is when you put an effort in writing an answer and then it cannot be posted because the question got closed. So I'm sorry that you couldn't post your answer but thank you that you posted it as an issue because I think this is a nice improvement over my code and people will benefit from reading it. I'm sure I'll add your suggestions to my examples. Meanwhile I edited my question to address every issue that people raised in the comments and to conform to every guideline that was referenced. I hope that it will get reopened and you'll be able to also post your answer there. Thanks.
